uxarray/grid/geometry.py

```diff
@@ -242,3 +249,113 @@ def _grid_to_matplotlib_linecollection(grid):
 
     # need transform? consider adding it later if needed
     return LineCollection(lines)
+
+
+def _pole_point_inside_polygon(pole, face_edge_cart):
+    """Determines if a pole point is inside a polygon.
+
+    .. note::
+        - If the pole point is on the edge of the polygon, it will be considered "inside the polygon".
+
+    Parameters
+    ----------
+    pole : str
+        Either 'North' or 'South'.
+    face_edge_cart : np.ndarray
+        A face polygon represented by edges in Cartesian coordinates. Shape: (n_edges, 2, 3)
+
+    Returns
+    -------
+    bool
+        True if pole point is inside polygon, False otherwise.
+
+    Raises
+    ------
+    ValueError
+        If the provided pole is neither 'North' nor 'South'.
+
+    Warning
+    -------
+    UserWarning
+        Raised if the face contains both pole points.
+    """
+    if pole not in POLE_POINTS:
+        raise ValueError('Pole point must be either "North" or "South"')
+
+    # Classify the polygon's location
+    location = _classify_polygon_location(face_edge_cart)
+    pole_point = POLE_POINTS[pole]
+    ref_point = np.array([1.0, 0.0, 0.0])  # Reference point on the equator
```

Comment:
I would consider declaring this at the top of the module, similar to the pole points as `EQUATOR`